docs/libvirt-node-networking.md

```diff
@@ -0,0 +1,243 @@
+# Layer two connectivity for nested k8s hosts
+
+The purpose of this proposal, is to allow connecting VMs to the node network,
+in contrast to
+[docs/libvirt-pod-networking.md](docs/libvirt-pod-networking.md), which allows
+VMs to be connected to the Kubernetes Overlay network. As a consequence there
+are three main points to consider:
+
+ 1) How and where to specify what the master host network device to connect to
+    is.
+ 3) How can the nested nodes and the real node communicate with each other.
+ 2) How to bring a device in position, so that libvirt can use it.
+
+## How to specify the host network device
+
+Right now, adding an extra network type `NodeNetwork` will be added. It does
+not take any extra arguments. On the node, the plugin which creates a new
+endpoint for the device will get the node device name passed in by
+virt-handler. Virt-handler, which has to invoke the plugin from the host
+network namespace view, will fetch the node IP from the `Node` object in the
+cluster, and derive the device from it.
+
+The section on the `Node` from which the IP needs to be taken:
+
+```yaml
+status:
+  addresses:
+  - address: 192.168.121.180
+    type: InternalIP
+```
+
+Based on this IP, the master device can be derived and passed in to the plugin.
+It should be sufficient to derive this information once, on virt-handler start.
+
+When a VM contains a network configuration like this:
+
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      interfaces:
+      - type: NodeNetwork
```

Comment:
would it make any more/less sense to call this 'HostNetwork' instead of 'NodeNetwork'?  

I only bring this up because pods call it host networking.